### PR TITLE
adding galaxy s7 exynos (flat and edge)

### PR DESCRIPTION
**Pull request by Kethen**
How long do artifacts lasts? I see the fp3 yml using that for shipping the recovery image.

How to make systemimage available to the installer?

Last CI run:
https://gitlab.com/ubports/porting/community-ports/android11/samsung-galaxy-s7/samsung-exynos8890/-/pipelines/597880645

### CODE REVIEW

**Comment by amartinz** [comment]
> How long do artifacts lasts?

I am not sure, maybe @Flohack74 knows?
In any case, with GitLabs upcoming policy change for projects and storage, we might need a different solution than relying on build artifacts?

Ultimatively we should host the images ourselves.

> How to make systemimage available to the installer?

This needs to be manually added, also @Flohack74 i guess 😄

**Comment by Kethen** [comment]
The installer config is working at the moment, a minimal boot image is made with https://github.com/Kethen/samsung-exynos8890/blob/main/src/reboot_recovery.c as init to reboot to recovery, because the recovery image is too big for the boot partition

make_ext4fs is not built with halium 11.0 recovery, while mkfs.ext4 is built, so adb:shell is used for wiping

Need help with recovery and boot hosting, currently hosted as a github release for testing

Need help with hero2lte testing because I don't own a S7 edge

Reboot to bootloader meanwhile requires `reboot download` instead of `reboot bootloader`, which is not supported by the installer's schema
https://github.com/Kethen/android_kernel_samsung_universal8890/blob/69caca3256508d80a417c64981a7a36f3c484930/drivers/staging/samsung/sec_reboot.c#L170

While I can change to kernel to boot to download mode with the bootloader command, I'm not sure if it would be a nicer solution to include 'download' in the adb:reboot schema

File: v2/devices/hero2lte.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          value: true
+      - actions:
+          - adb:shell:
+              args: ["mount", "/data"]
+        condition:
+          var: "wipe"
+          value: true
+      - actions:
+          - adb:shell:
+              args: ["mkdir", "/data/cache"]
```
Comment (amartinz) [inline v2/devices/hero2lte.yml:88]: The system should be able to detect and create that itself instead of relying on the installer, shouldn't it?

**Comment by amartinz** [comment]
Also your usage of raw `adb:shell` commands requires `compatible_installer: ">=0.9.6-beta"`.

File: v2/devices/hero2lte.yml (hunk 1 above)
Comment (Kethen) [reply v2/devices/hero2lte.yml:88]: That is only for after wiping, guarded by the "wipe" var

**Comment by Kethen** [comment]
updated `compatible_installer` to `">=0.9.6-beta"`

File: v2/devices/hero2lte.yml (hunk 1 above)
Comment (amartinz) [reply v2/devices/hero2lte.yml:88]: Yes, that's what i mean :)

Init should check for its existence and recreate it, if it does not exist, instead of the installer.

Comment (Kethen) [reply v2/devices/hero2lte.yml:88]: Hmm.. I'm probably missing something then on the recovery

I didn't find a way to write the fstab so that `ensure_path_mounted` would preform mounting /data and creating /data/cache for me with just /cache mount, so these modifications were made to redirect the cache mount with a symbolic link, and create /data/cache if needed on recovery start

https://github.com/Kethen/halium_bootable_recovery/commit/51eb0798308af2a8db61d836bff51e7f2a7b2ef4

/cache mount was disabled in fstab meanwhile

https://github.com/Kethen/android_device_samsung_universal8890-common/blob/lineage-18.1-ut-recovery/ramdisk/etc/fstab.samsungexynos8890

That's why in the installer config, after the wipe with mkfs.ext4, the next step is to re-create /data/cache before `systemimage:install:`

**Comment by SpectreDev-007** [comment]
Hero2lte installer works fine

**Comment by amartinz** [comment]
@Kethen is this good to go?
Thanks for the work you put in!

**Comment by Kethen** [comment]
@amartinz I think so, thanks for reviewing
